### ISSUE TITLE
"make dir" doesn't work on windows

### ISSUE THREAD

**Issue by NanoUser3000** [Sobeston/ziglearn#167]
The "make dir" test fails with EBUSY on deleteTree() call `  std.fs.cwd().deleteTree("test-tmp") catch unreachable;`.
I should mention that I'm on windows with wsl2 installed, running tests in git bash. Everything up to this test worked fine.

Closing directory first by adding `iter_dir.close();` to defer block (and Changing iter_dir from `const` to `var`) results in program hanging - it doesn't exit deleteTree() call (snippet with those changes below).

In summary: it doesn't work and I am not sure how to fix it.

Change I tried (snippet that doesn't work):
   ```
var iter_dir = try std.fs.cwd().openIterableDir(
    "test-tmp",
     .{},
);
defer {
    iter_dir.close();
    std.fs.cwd().deleteTree("test-tmp") catch unreachable;
}
```

**Comment by NanoUser3000** [comment]
My version was 
```
$ zig version
0.11.0-dev.2985+3f3b1a680
```
I hear it works with modification I mentioned on, for example,  `0.11.0-dev.1491+2b27bc2c6`, but hangs on my version and latest `dev.3000`

It might be an example being outdated example and a bug in Zig as well. Or is just very platform dependent.

**Comment by Sobeston** [comment]
I believe  this has been fixed; reopen if the problem persists
